tests/returntypes/jax/test_jax_jit_qnode_new.py

```diff
@@ -2056,3 +2040,267 @@ def circuit(a):
 
         assert isinstance(jac[1], jax.numpy.ndarray)
         assert jac[1].shape == (4, 2)
+
+
+hessian_fn = [
+    jax.hessian,
+    lambda fn, argnums=0: jax.jacrev(jax.jacfwd(fn, argnums=argnums), argnums=argnums),
+    lambda fn, argnums=0: jax.jacfwd(jax.jacrev(fn, argnums=argnums), argnums=argnums),
+]
+
+
+@pytest.mark.parametrize("dev_name,diff_method,mode", qubit_device_and_diff_method_and_mode)
+@pytest.mark.parametrize("shots", [None, 10000])
+@pytest.mark.parametrize("hessian", hessian_fn)
+class TestReturnHessian:
+    """Class to test the shape of the Hessian with different return types."""
+
+    def test_hessian_expval_multiple_params(self, dev_name, diff_method, hessian, shots, mode):
+        """The hessian of single a measurement with multiple params return a tuple of arrays."""
+        dev = qml.device(dev_name, wires=2)
+
+        if diff_method == "adjoint":
+            pytest.skip("Test does not supports adjoint because second order diff.")
+
+        par_0 = jax.numpy.array(0.1)
+        par_1 = jax.numpy.array(0.2)
+
+        @qnode(dev, interface="jax", diff_method=diff_method, max_diff=2)
```

Comment:
Is there a difference between using `interface="jax-jit"` directly in the QNode and using `interface="jax"` in the QNode but decorating the hessian function with `jax.jit`? In the changelog we have an example with the former, but these tests do the latter.

Reply:
[With `interface="jax"` PennyLane attempts to determine](https://github.com/PennyLaneAI/pennylane/blob/master/pennylane/interfaces/jax.py#L30) if the JAX-Python or the JAX-JIT interface should be used. The implementation works in a way, that `interface="jax"` and decorating a function with `jax.jit` will resolve to JAX-JIT.

When specifying `interface="jax-python"`/`interface="jax-jit"`, we definitely use the JAX-Python/JAX-JIT interface.

This is not a purely functional solution and is discouraged by JAX devs. Nonetheless, it brings user convenience to our users.